### PR TITLE
Add missing color design tokens

### PR DESCRIPTION
**Pull request by ahmad-demos**
**Ticket:** https://yaits.demos-deutschland.de/T30974

This PR establishes an almost complete set of **color design tokens** within demosplan-ui.

Related PR within demosplan-core: https://github.com/demos-europe/demosplan-core/pull/898

:warning: **Want to test it locally?**

- `yarn install`
- yarn build-tokens`

Within the `tokens/scss` dir, there should be several new files.

#### New Hex Colors

- Palettes are supplemented with the new color palette `teal`
- The `magenta` palette has been refined to also have 7 steps, values have been adapted to form a more regular curve
- The value of `color.palette.yellow.dark-1` has been adapted to be slightly darker, to better match the yellow palette curve
- The value of `color.palette.red.base` has been tweaked to be slightly less aggressive

<details>
<summary>See all palette colors</summary>

image

```scss
$dp-color-magenta-dark-2: #650324 !default;
$dp-color-magenta-dark-1: #960739 !default;
$dp-color-magenta-base: #d4004b !default;
$dp-color-magenta-light-1: #e73e79 !default;
$dp-color-magenta-light-2: #ff9fbd !default;
$dp-color-magenta-light-3: #fbdbe5 !default;
$dp-color-magenta-light-4: #fdedf2 !default;
$dp-color-purple-dark-2: #4b0085 !default;
$dp-color-purple-dark-1: #5e00a6 !default;
$dp-color-purple-base: #8f4fc0 !default;
$dp-color-purple-light-1: #a572cd !default;
$dp-color-purple-light-2: #bf99db !default;
$dp-color-purple-light-3: #dfcced !default;
$dp-color-purple-light-4: #f3ecf8 !default;
$dp-color-yellow-dark-2: #845d01 !default;
$dp-color-yellow-dark-1: #d09200 !default;
$dp-color-yellow-base: #ffbe24 !default;
$dp-color-yellow-light-1: #ffd87c !default;
$dp-color-yellow-light-2: #ffe5a7 !default;
$dp-color-yellow-light-3: #ffefca !default;
$dp-color-yellow-light-4: #fff7e5 !default;
$dp-color-green-dark-2: #064006 !default;
$dp-color-green-dark-1: #186218 !default;
$dp-color-green-base: #228b22 !default;
$dp-color-green-light-1: #5cba5c !default;
$dp-color-green-light-2: #93e293 !default;
$dp-color-green-light-3: #def7de !default;
$dp-color-green-light-4: #effbef !default;
$dp-color-teal-dark-2: #004f5d !default;
$dp-color-teal-dark-1: #007286 !default;
$dp-color-teal-base: #0095b0 !default;
$dp-color-teal-light-1: #29acc4 !default;
$dp-color-teal-light-2: #66cee2 !default;
$dp-color-teal-light-3: #bcedf6 !default;
$dp-color-teal-light-4: #e6fbff !default;
$dp-color-red-dark-2: #820000 !default;
$dp-color-red-dark-1: #b20000 !default;
$dp-color-red-base: #e52222 !default;
$dp-color-red-light-1: #fc5e5e !default;
$dp-color-red-light-2: #ffb3b3 !default;
$dp-color-red-light-3: #fcdddd !default;
$dp-color-red-light-4: #fff0f0 !default;
$dp-color-blue-dark-2: #004078 !default;
$dp-color-blue-dark-1: #005eb1 !default;
$dp-color-blue-base: #006fd0 !default;
$dp-color-blue-light-1: #44a8ff !default;
$dp-color-blue-light-2: #82c5ff !default;
$dp-color-blue-light-3: #cee6fb !default;
$dp-color-blue-light-4: #ebf5ff !default;
$dp-color-neutral-dark-2: #333333 !default;
$dp-color-neutral-dark-1: #595959 !default;
$dp-color-neutral-base: #757575 !default;
$dp-color-neutral-light-1: #aeaeae !default;
$dp-color-neutral-light-2: #d3d3d3 !default;
$dp-color-neutral-light-3: #ebe9e9 !default;
$dp-color-neutral-light-4: #f5f5f5 !default;
$dp-color-white: #ffffff !default;
$dp-color-black: #000000 !default;
```
</details>

#### New Brand Tokens

3 pairs of brand tokens are introduced: `main`, `alt` and `hightlight` - each with their `contrast` counterpart holding a color value that ensures accessibility when a pair is used as foreground/background color.

<details>
<summary>See all brand colors</summary>

```scss
$dp-color-highlight-contrast: $dp-color-white !default;
$dp-color-highlight: $dp-color-red-dark-1 !default;
$dp-color-alt-contrast: $dp-color-white !default;
$dp-color-alt: $dp-color-teal-base !default;
$dp-color-main-contrast: $dp-color-white !default;
$dp-color-main: $dp-color-blue-base !default;
```
</details>

#### New UI Tokens

The following alias token families have been added to demosplan-ui:

- interactive
- message
- status
- text
- selection
- border
- background

A more detailed, yet incomplete, description of each token can be found in the `tokens.color.stories.mdx` file. A thorough description can be found in the `$description` field of each token within the token definition in `color.ui.json`.

<details>
<summary>See all ui colors</summary>

```scss
$dp-color-background-medium: $dp-color-neutral-light-3 !default;
$dp-color-background-light: $dp-color-neutral-light-4 !default;
$dp-color-background-default: $dp-color-white !default;
$dp-color-border-default: $dp-color-neutral-light-2 !default;
$dp-color-selection-bg: $dp-color-blue-light-3 !default;
$dp-color-selection-text: $dp-color-blue-dark-1 !default;
$dp-color-text-deleted-bg-muted: $dp-color-yellow-light-3 !default;
$dp-color-text-deleted-bg: $dp-color-yellow-light-2 !default;
$dp-color-text-inserted-bg-muted: $dp-color-blue-light-4 !default;
$dp-color-text-inserted-bg: $dp-color-blue-light-3 !default;
$dp-color-text-marked-bg: $dp-color-blue-light-3 !default;
$dp-color-text-marked: $dp-color-blue-dark-2 !default;
$dp-color-text-on-dark: $dp-color-white !default;
$dp-color-text-muted: $dp-color-neutral-base !default;
$dp-color-text-default: $dp-color-neutral-dark-2 !default;
$dp-color-status-neutral-text: $dp-color-neutral-dark-1 !default;
$dp-color-status-neutral-fill: $dp-color-neutral-base !default;
$dp-color-status-complete-text: $dp-color-green-dark-1 !default;
$dp-color-status-complete-fill: $dp-color-green-base !default;
$dp-color-status-failed-text: $dp-color-red-dark-1 !default;
$dp-color-status-failed-fill: $dp-color-red-base !default;
$dp-color-status-failed-border: $dp-color-red-base !default;
$dp-color-status-changed-text: $dp-color-yellow-dark-2 !default;
$dp-color-status-changed-fill: $dp-color-yellow-dark-1 !default;
$dp-color-status-progress-text: $dp-color-blue-dark-1 !default;
$dp-color-status-progress-fill: $dp-color-blue-base !default;
$dp-color-message-debug-text: $dp-color-purple-dark-2 !default;
$dp-color-message-debug-interactive: $dp-color-purple-dark-1 !default;
$dp-color-message-debug-fill: $dp-color-purple-dark-1 !default;
$dp-color-message-debug-border: $dp-color-purple-light-1 !default;
$dp-color-message-debug-bg: $dp-color-purple-light-3 !default;
$dp-color-message-success-text: $dp-color-green-dark-2 !default;
$dp-color-message-success-interactive: $dp-color-green-base !default;
$dp-color-message-success-fill: $dp-color-green-dark-1 !default;
$dp-color-message-success-border: $dp-color-green-light-1 !default;
$dp-color-message-success-bg: $dp-color-green-light-3 !default;
$dp-color-message-severe-text: $dp-color-red-dark-2 !default;
$dp-color-message-severe-interactive: $dp-color-red-dark-1 !default;
$dp-color-message-severe-fill: $dp-color-red-dark-1 !default;
$dp-color-message-severe-border: $dp-color-red-light-1 !default;
$dp-color-message-severe-bg: $dp-color-red-light-3 !default;
$dp-color-message-warning-text: $dp-color-yellow-dark-2 !default;
$dp-color-message-warning-interactive: $dp-color-yellow-dark-2 !default;
$dp-color-message-warning-fill: $dp-color-yellow-dark-2 !default;
$dp-color-message-warning-border: $dp-color-yellow-base !default;
$dp-color-message-warning-bg: $dp-color-yellow-light-3 !default;
$dp-color-message-info-text: $dp-color-blue-dark-2 !default;
$dp-color-message-info-interactive: $dp-color-blue-base !default;
$dp-color-message-info-fill: $dp-color-blue-dark-1 !default;
$dp-color-message-info-border: $dp-color-blue-light-1 !default;
$dp-color-message-info-bg: $dp-color-blue-light-3 !default;
$dp-color-interactive-muted: $dp-color-neutral-light-1 !default;
$dp-color-interactive-active: $dp-color-blue-dark-2 !default;
$dp-color-interactive-hover-subtle: $dp-color-neutral-light-1 !default;
$dp-color-interactive-hover: $dp-color-blue-dark-1 !default;
$dp-color-interactive-default: $dp-color-blue-base !default;
```
</details>

#### New Internal "Status" tokens

While sparely used at the moment, status tokens are introduced to be used internally. With status tokens, we have the possibility to mark tokens as In Review, Approved or Deprecated. Token status may be rendered in the docs, later on.

### New Scss variables structure

When running `yarn build-tokens`, Scss variables for colors do not contain the "domain" part (ui, palette eg...). We discussed this with the team, and agreed on the variables being expressive enough to drop the "domain" part and shorten variable length a little.

### New File Structure

Palette, Data, Brand, and UI tokens are defined within their own json files. The generated Scss
files also follow that structure.

For internal tokens, a new `_.json` File is created.

### Documentation

To document the basics, a tokens overview and a color tokens article are established. The latter still leaves out some UI tokens that still have to be documented. As it is planned to generate color tokens on an automated base, this is left out intentionally for now.

### CODE REVIEW

File: tokens/tokens.color.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,145 @@
+import { Meta } from '@storybook/addon-docs'
```
Comment (hwiem) [inline tokens/tokens.color.stories.mdx:1]: Thanks for adding this (really good) documentation!

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,11 @@
 # Changelog
 
 ## UNRELEASED
+
+### Changed
+
+- **Breaking**: Add color design tokens, split generated Scss into multiple files ([#108](https://github.com/demos-europe/demosplan-ui/pull/108))
```
[the remark this answers is not in the document]
Comment (spiess-demos) [reply CHANGELOG.md]: added in 41057baf86add903ad5536762ac2b75a2dafc70c

File: tokens/color/color.ui.json [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+            "value": "{color.palette.blue.base}",
+            "themeable": true
+          },
+          "text": {
+            "value": "{color.palette.blue.dark-2}",
+            "themeable": true
+          }
+        },
+        "warning": {
+          "$description": "Use these colors to noticeable communicate uncritical information.",
```
[the remark this answers is not in the document]
Comment (spiess-demos) [reply tokens/color/color.ui.json]: thanks, you're right! 41057baf86add903ad5536762ac2b75a2dafc70c

File: tokens/tokens._overview.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+## Naming scheme
+
+Tokens are made up of the following scheme:
+
+```
+<Namespace>-<Category>(-<Domain>)?-<Concept/Element>-<Variant/State>(-<Context>)?
+```
+
+*Namespace* is always "dp", *Category* being "color", "space", or "font-size", *Domain* being "brand", "data", "ui"
```
[the remark this answers is not in the document]
Comment (spiess-demos) [reply tokens/tokens._overview.stories.mdx]: right! formatted as a list in 41057baf86add903ad5536762ac2b75a2dafc70c

File: tokens/color/color.brand.json [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        "value": "{color.palette.teal.base}",
+        "themeable":  true
+      },
+      "alt-contrast": {
+        "$description": "Should pass AAA contrast check with the value of 'alt'.",
+        "value": "{color.palette.white}",
+        "themeable":  true
+      },
+      "highlight": {
+        "$description": "Brand accent color used for communicating active or focused state of interactive elements.",
```
Comment (hansspiess) [inline tokens/color/color.brand.json:26]: This is simply not right anymore. 
```suggestion
        "$description": "Brand accent color to be used sparingly to raise awareness for important elements.",
```

File: src/components/core/DpBulkEditHeader.vue [hunk 6]
```diff
@@ -1,7 +1,7 @@
 <template>
   <div
     role="menu"
-    class="layout__item u-12-of-12 u-p-0_5 bg-color--system-info-light-1 line-height--1_6">
+    class="layout__item u-12-of-12 u-p-0_5 bg-color-ui-selection line-height--1_6">
```
Comment (salisdemos) [inline src/components/core/DpBulkEditHeader.vue:4]: As You point me to think about it @spiess-demos I actually would omit the `ui-` but this is just my personal feeling.

**Review by salisdemos — APPROVED** [review]

File: src/utils/lengthHint/lengthHint.js [hunk 7]
```diff
@@ -11,7 +11,7 @@ const exactlengthHint = (currentLength, requiredLength) => {
     ? Translator.trans('input.text.exactlength.short', {
       requiredlength: requiredLength,
       count: currentLength,
-      class: (currentLength === reqLength) ? 'color-ui-info' : 'color-ui-error'
+      class: (currentLength === reqLength) ? 'color-ui-status-progress-text' : 'color-ui-status-failed-text'
```
[the remark this answers is not in the document]
Comment (spiess-demos) [reply src/utils/lengthHint/lengthHint.js:14]: You are technically right! However as we added some new design tokens, this is one of the cases where a newly added token fits better than the vaguely generic `color-ui-info`. While the character counting number will have the color for "progress", applied to text, if the character limit is not exceeded, the color for "failed" is used if exceeded.

File: src/utils/lengthHint/lengthHint.js [hunk 8]
```diff
@@ -34,7 +34,7 @@ const maxlengthHint = (currentLength, maxlength) => {
     ? Translator.trans('input.text.maxlength.short', {
       max,
       count: max - currentLength,
-      class: (max - currentLength > errorThreshold) ? 'color-ui-info' : 'color-ui-error'
+      class: (max - currentLength > errorThreshold) ? 'color-ui-status-progress-text' : 'color-ui-status-failed-text'
```
[the remark this answers is not in the document]
Comment (spiess-demos) [reply src/utils/lengthHint/lengthHint.js:37]: See above.

File: src/utils/lengthHint/lengthHint.js [hunk 9]
```diff
@@ -52,7 +52,7 @@ const minlengthHint = (currentLength, minlength) => {
     ? Translator.trans('input.text.minlength.short', {
       min: min,
       count: min - currentLength,
-      class: (min <= currentLength) ? 'color-ui-info' : 'color-ui-error'
+      class: (min <= currentLength) ? 'color-ui-status-progress-text' : 'color-ui-status-failed-text'
```
[the remark this answers is not in the document]
Comment (spiess-demos) [reply src/utils/lengthHint/lengthHint.js:55]: See above

**Review by sakutademos — APPROVED** [review]
lgtm! code review

File: src/components/core/DpBulkEditHeader.vue (hunk 6 above)
Comment (spiess-demos) [reply src/components/core/DpBulkEditHeader.vue]: I updated utility classes in https://github.com/demos-europe/demosplan-core/pull/1326 and here, in b2683da3a95a7c6897591616f466078647654bd1

**Review by salisdemos — APPROVED** [review]

File: CHANGELOG.md [hunk 10]
```diff
@@ -18,6 +18,10 @@
 
 - ([#175](https://github.com/demos-europe/demosplan-ui/pull/175)) adjust layout of DpPager ([@muellerdemos](https://github.com/muellerdemos))
 
+### Changed
+
+- **Breaking**: ([#108](https://github.com/demos-europe/demosplan-ui/pull/108)) Add color design tokens, split generated Scss into multiple files ([@spiess-demos](https://github.com/spiess-demos))
```
Comment (spiess-demos) [inline CHANGELOG.md:23]:
```suggestion
- **Breaking**: ([#108](https://github.com/demos-europe/demosplan-ui/pull/108)) Add color design tokens, split generated Scss into multiple files ([@ahmad-demos](https://github.com/ahmad-demos), [@spiess-demos](https://github.com/spiess-demos))
```

**Pull request merged**